frontend/packages/console-dynamic-plugin-sdk/scripts/generate-pkg-assets.ts

```diff
@@ -3,31 +3,41 @@ import * as fs from 'fs-extra';
 import * as _ from 'lodash';
 import * as readPkg from 'read-pkg';
 import { resolvePath, relativePath } from './utils/path';
+import { getCorePackage, getInternalPackage, getWebpackPackage } from './package-definitions';
 
-const createPackageJson = (packagePath: string) => {
-  const packageJson = readPkg.sync({ normalize: false });
-  packageJson.name = '@openshift-console/dynamic-plugin-sdk';
-  delete packageJson.private;
-  packageJson.license = 'Apache-2.0';
-  packageJson.main = 'lib/index-lib.js';
-  packageJson.readme = 'README.md';
-  packageJson.peerDependencies = _.pick(packageJson.devDependencies, 'webpack');
-  delete packageJson.dependencies;
-  delete packageJson.devDependencies;
-  delete packageJson.scripts;
-  fs.writeFileSync(packagePath, JSON.stringify(packageJson, null, 2));
+const writePackageManifest = (manifest: readPkg.PackageJson, outDir: string) => {
```

Comment:
I see we were already doing this, but I'm curious what the benefit of generating package.json is compared to simply having 3 packages in the repo with their own static package.json files.

Reply:
This is mainly about computing the right `dependencies` in the generated package manifest.

- `core` and `internal` SDK packages have dependency names taken from shared plugin modules (`shared-modules.ts`) and their semver ranges parsed from root `package.json`
- `webpack` SDK package has dependency names declared explicitly in `package-definitions.ts` and their semver ranges parsed from both root and dynamic SDK `package.json`

What we can do is a combination of both, have 3 static `package.json` files which we then augment with dependencies.